### ISSUE TITLE
Question about catalog views and HAL

### ISSUE THREAD

**Issue by dcreager** [mnichols/consistency#1]
Ha, hopefully this isn't a strange way to ask a question!

In the caching doc, you have this for a catalog view:

``` js
var AssetsCatalog = {
    _links: {
        self: <Link>
        ,default_ordering: <Link>
    }
    catalog: {
        '/api/assets/123': {
            name: <String>
        }
        ,'/api/assets/234' {
            name: <String>
        }
        ,...
    }
}
```

For the most part, that makes lots of sense.  But what if the mutable state of a resource also includes a link?  My understanding of HAL is that all of the links must appear in the `_links` section, and I can't see a way to reproduce the "hash with URLs as keys" that you're using down below for the simple state values.

**Comment by mnichols** [comment]
The `catalog` is not a reproduction of `_links`. Instead it is an index that happens to use resource uris as their keys. The idea of traversing to these resources is not the intent, but rather to encapsulate the cataloging of all assets along with useful (small bits) of metadata . The important thing is you do not want to reproduce that metadata (eg `name`).

**Comment by dcreager** [comment]
Right, but with if that metadata contains a link?  An asset has a name and also an owner, and that owner can change (which is why we're putting it here in the mutable catalog view and not in the immutable view):

``` js
var AssetsCatalog = {
  catalog: {
    '/api/assets/123': {
      name: "Cool name",
      owner: "/api/users/dcreager"
    }
  }
}
```

**Comment by mnichols** [comment]
Would you PUT something like this to change the owner?: 

``` js
{
    catalog: {
        '/api/assets/123': {
             name: 'NewName',
             owner: '/api/users/newOwnerUrl'
        } 
    }
}
```

**Comment by dcreager** [comment]
Yep, that's what I figured would be the pattern.  (Or PATCH so you could leave out the `name` field)

**Comment by mnichols** [comment]
This really isn't the place for a link like this unless you are cataloging URIs. To change the owner, one would likely traverse to the asset referenced in the key and then following links from there. 
Instead you'd expose the ownerName as a kind of projection . It is reasonable to have out-of-band docs that indicate you wouldnt update owner there (`400` if attempted), but that is less desirable than splitting up attrs to avoid transaction considerations.

Just because a URI is being used though, it isn't necessarily a HAL link. In other words, we traverse using the `_links` collection but also treat the url as an UUID for applicative purposes.

**Comment by mnichols** [comment]
Ok imagine a client downloading this catalog _once_.  Assuming this catalog isn't that volatile, the client can hold a reference to that  resource. 
Now imagine that all the resources that are keyed in that catalog have a far more aggresive caching approach. The burden on the server _and_ the client is considerably less when accessing each referenced resource in turn.
